### PR TITLE
Add support for Grafana alerts in v9

### PR DESCRIPTION
**Pull request by JamesGibo**
Had to add a separate class for v8 and v9 alerts as the format was quite different, this meant I have renamed the AlertRule class to  
Changes the name of `AlertRule` to `AlertRulev8` and `AlertRulev9`.

### CODE REVIEW

**Comment by JamesGibo** [comment]
Still need to add some documentation and possible add some more input validation, but @OscarVanL would you be able to test this version with you v8 alerting rules and check that I have not broken anything for you other than the name of the `AlertRule` class which you will have to change to `AlertRulev8`

**Comment by JamesGibo** [comment]
> Were you hoping to get this change in before making a new release?

Yes I was, so we don't have to make a breaking change of the class name after it has been released

**Comment by JamesGibo** [comment]
@dholbach Not sure if you have time to review but would appreciate a second pair of eyes on the code and any other suggestions on to include support for Grafana v9 alerts nicely with Grafana v8 alerts.

**Comment by JamesGibo** [comment]
> (will only get around to trying this next week though)

@OscarVanL if you do have time to test it that would be much apricated as I don't have any Grafana v8 alerts to test with.

I have finished my update to the code and welcome any feedback.

**Comment by OscarVanL** [comment]
> > (will only get around to trying this next week though)
> 
> 
> 
> @OscarVanL if you do have time to test it that would be much apricated as I don't have any Grafana v8 alerts to test with.
> 
> 
> 
> I have finished my update to the code and welcome any feedback. 
> 
> 

So sorry James, I didn't get around to trying this out, and now I'm on holiday for the next 2.5 weeks 🙈

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
+  pull_request:
 
 jobs:
   build-n-publish:
     name: Run tests
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-20.04
     strategy:
       matrix:
         python: ['3.6', '3.7', '3.8', '3.9', '3.10']
```
Comment (dholbach) [inline .github/workflows/run-tests.yml:14]:
```suggestion
        python: ['3.7', '3.8', '3.9', '3.10']
```

**Comment by dholbach** [comment]
I can't review it in depth, I'm afraid, but from skimming over it, it LGTM.

Will this constitute a breaking change? Shall we bump the minor for the next release?

**Review by dholbach — APPROVED** [review]

**Comment by JamesGibo** [comment]
> I can't review it in depth, I'm afraid, but from skimming over it, it LGTM.
> 
> Will this constitute a breaking change? Shall we bump the minor for the next release?

No worries.

Not a breaking change really as the the code for alerts in Grafana v8 was only just added and not release yet. 
But was thinking of doing a minor release anyway as adding alerting support is a significant feature.

**Comment by williamtaylor99** [comment]
Hi @JamesGibo , I'm working with @OscarVanL on our alerts. I've tested the V8 rules and unfortunately it doesn't seem to quite work. I believe this is because of the `AlertGroup` requiring a folder which is a Grafana 9.X attribute. I get the error `failed to update rule group: failed to add rules: a conflicting alert rule is found: rule title under the same organisation and folder should be unique` from the API, even though no rules have the same name. Making this argument optional will most likely fix this as this is the only diff in our generated JSON.

**Comment by JamesGibo** [comment]
> Hi @JamesGibo , I'm working with @OscarVanL on our alerts. I've tested the V8 rules and unfortunately it doesn't seem to quite work. I believe this is because of the `AlertGroup` requiring a folder which is a Grafana 9.X attribute. I get the error `failed to update rule group: failed to add rules: a conflicting alert rule is found: rule title under the same organisation and folder should be unique` from the API, even though no rules have the same name. Making this argument optional will most likely fix this as this is the only diff in our generated JSON.

Interesting, I presumed Grafana would just ignore any extra fields as it does with dashboards. 
@williamtaylor99  have you tried manually removing the `folder` from the generated JSON to check this is the cause of the issue and not a miss configuration?

**Comment by williamtaylor99** [comment]
> > Hi @JamesGibo , I'm working with @OscarVanL on our alerts. I've tested the V8 rules and unfortunately it doesn't seem to quite work. I believe this is because of the `AlertGroup` requiring a folder which is a Grafana 9.X attribute. I get the error `failed to update rule group: failed to add rules: a conflicting alert rule is found: rule title under the same organisation and folder should be unique` from the API, even though no rules have the same name. Making this argument optional will most likely fix this as this is the only diff in our generated JSON.
> 
> Interesting, I presumed Grafana would just ignore any extra fields as it does with dashboards. @williamtaylor99 have you tried manually removing the `folder` from the generated JSON to check this is the cause of the issue and not a miss configuration?

Ah we had some incorrect json committed recently, this works and looks good to me :smile:

**Comment by JamesGibo** [comment]
> > > Hi @JamesGibo , I'm working with @OscarVanL on our alerts. I've tested the V8 rules and unfortunately it doesn't seem to quite work. I believe this is because of the `AlertGroup` requiring a folder which is a Grafana 9.X attribute. I get the error `failed to update rule group: failed to add rules: a conflicting alert rule is found: rule title under the same organisation and folder should be unique` from the API, even though no rules have the same name. Making this argument optional will most likely fix this as this is the only diff in our generated JSON.
> > 
> > 
> > Interesting, I presumed Grafana would just ignore any extra fields as it does with dashboards. @williamtaylor99 have you tried manually removing the `folder` from the generated JSON to check this is the cause of the issue and not a miss configuration?
> 
> Ah we had some incorrect json committed recently, this works and looks good to me 😄

Great, thanks for checking. 

I will look at merging and making a release in the next week.

**Comment by OscarVanL** [comment]
Nice work 😃 we just updated to Grafana 9 so this is great timing!

**Pull request merged**